From b8dc0eeb2fb6c49ae0ac9cbc86bd20c0bf3bdee7 Mon Sep 17 00:00:00 2001
From: Vadim Zeitlin <vadim@wxwidgets.org>
Date: Thu, 30 Nov 2023 03:02:13 +0100
Subject: [PATCH] Fix wxGLCanvasEGL compilation when using X11 without Wayland

gs_alreadySetSwapInterval variable is used for both X11 and Wayland and
so shouldn't be defined only inside GDK_WINDOWING_WAYLAND check.

Simply remove the check to fix the problem.

Also add an explanatory comment.

Closes #24076.
---
 src/unix/glegl.cpp | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

--- src/unix/glegl.cpp.orig
+++ src/unix/glegl.cpp
@@ -43,8 +43,9 @@
 
 static const char* TRACE_EGL = "glegl";
 
-#ifdef GDK_WINDOWING_WAYLAND
-
+// We can't add a member variable to wxGLCanvasEGL in 3.2 branch, so emulate it
+// by encoding the corresponding boolean value via the presence of "this"
+// pointer in the given hash set.
 #include "wx/hashset.h"
 
 namespace
@@ -58,8 +59,6 @@ wxGLCanvasSet gs_alreadySetSwapInterval;
 
 } // anonymous namespace
 
-#endif // GDK_WINDOWING_WAYLAND
-
 // ----------------------------------------------------------------------------
 // wxGLContextAttrs: OpenGL rendering context attributes
 // ----------------------------------------------------------------------------
@@ -633,9 +632,9 @@ wxGLCanvasEGL::~wxGLCanvasEGL()
     DestroyWaylandSubsurface();
     g_clear_pointer(&m_wlEGLWindow, wl_egl_window_destroy);
     g_clear_pointer(&m_wlSurface, wl_surface_destroy);
+#endif
 
     gs_alreadySetSwapInterval.erase(this);
-#endif
 }
 
 void wxGLCanvasEGL::CreateWaylandSubsurface()
